cmd/run/cmd.go

```diff
@@ -120,50 +119,24 @@ func parseConfigFromFlags() error {
 			return fmt.Errorf("invalid COA private key: %w", err)
 		}
 		cfg.COAKey = pkey
-	} else if keysPath != "" {
-		raw, err := os.ReadFile(keysPath)
-		if err != nil {
-			return fmt.Errorf("could not read the file containing list of keys for key-rotation mechanism, check if coa-key-file specifies valid path: %w", err)
-		}
-		var keysJSON []string
-		if err := json.Unmarshal(raw, &keysJSON); err != nil {
-			return fmt.Errorf("could not parse file containing the list of keys for key-rotation, make sure keys are in JSON array format: %w", err)
-		}
-
-		cfg.COAKeys = make([]crypto.PrivateKey, len(keysJSON))
-		sigAlgo := crypto.StringToSignatureAlgorithm(keyAlg)
-		if sigAlgo == crypto.UnknownSignatureAlgorithm {
-			return fmt.Errorf("invalid signature algorithm: %s", keyAlg)
-		}
-		for i, k := range keysJSON {
-			pk, err := crypto.DecodePrivateKeyHex(sigAlgo, k)
-			if err != nil {
-				return fmt.Errorf("a key from the COA key list file is not valid, key %s, error: %w", k, err)
-			}
-			cfg.COAKeys[i] = pk
-		}
-	} else if cloudKMSKeys != "" {
+	} else if cloudKMSKey != "" {
 		if cloudKMSProjectID == "" || cloudKMSLocationID == "" || cloudKMSKeyRingID == "" {
 			return fmt.Errorf(
 				"using coa-cloud-kms-keys requires also coa-cloud-kms-project-id & coa-cloud-kms-location-id & coa-cloud-kms-key-ring-id",
 			)
 		}
 
-		kmsKeys := strings.Split(cloudKMSKeys, ",")
-		cfg.COACloudKMSKeys = make([]flowGoKMS.Key, len(kmsKeys))
-		for i, key := range kmsKeys {
-			// key has the form "{keyID}@{keyVersion}"
-			keyParts := strings.Split(key, "@")
-			if len(keyParts) != 2 {
-				return fmt.Errorf("wrong format for Cloud KMS key: %s", key)
-			}
-			cfg.COACloudKMSKeys[i] = flowGoKMS.Key{
-				ProjectID:  cloudKMSProjectID,
-				LocationID: cloudKMSLocationID,
-				KeyRingID:  cloudKMSKeyRingID,
-				KeyID:      keyParts[0],
-				KeyVersion: keyParts[1],
-			}
+		// key has the form "{keyID}@{keyVersion}"
```

Comment:
**Note:** There's really no need to use one Cloud KMS key per Flow account key, and this might be actually costly to do so. Given that we might need ~1000 Flow account keys, we simply re-use one and the same Cloud KMS key for all of them.